docs/docsite/_ext/build_context.py

```diff
@@ -0,0 +1,110 @@
+"""Sphinx extension for setting up the build settings."""
+
+import subprocess
+from dataclasses import dataclass
+from functools import cache
+from pathlib import Path
+from tomllib import loads as parse_toml_string
+from typing import Literal
+
+from sphinx.application import Sphinx
+from sphinx.util import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+DOCSITE_ROOT_DIR = Path(__file__).parents[1].resolve()
+DOCSITE_EOL_CONFIG_PATH = DOCSITE_ROOT_DIR / 'end_of_life.toml'
+
+
+@dataclass(frozen=True)
+class Distribution:
+    end_of_life: list[str]
+    supported: list[str]
+
+    @classmethod
+    def from_dict(cls, raw_dist: dict[str, list[str]]) -> 'Distribution':
+        return cls(
+            **{
+                kind.replace('-', '_'): versions
+                for kind, versions in raw_dist.items()
+            },
+        )
+
+
+EOLConfigType = dict[str, Distribution]
+
+
+@cache
+def _read_eol_data() -> EOLConfigType:
+    raw_config_dict = parse_toml_string(DOCSITE_EOL_CONFIG_PATH.read_text())
```

Comment:
Not that the file is so large, but why not use the usual `tomllib.load` pattern instead of loading the whole text into memory first?

Reply:
Yeah, I also thought that it's probably not worth it.